src/components/comment.jsx

```diff
@@ -0,0 +1,144 @@
+        image: {
+          png: '',
+          webp: '',
+        },
+        username: '',
+      },
+    };
+    let newComments = props.comments.map((comment, index) => {
+      if (comment.id === props.comment.id) {
+        comment.replies.push(replyData);
+      }
+      return comment;
+    });
+
+    props.setComments(newComments);
+    setReplyClicked(false);
+    setReply('');
+  };
+
+  const handlePlus = () => {
+    const newComments = props.comments.map((comment, index) => {
+      if (comment.id === props.comment.id) {
+        comment.score += 1;
+      }
+      return comment;
+    });
+    props.setComments(newComments);
+  };
+
+  const handleMinus = () => {
+    const newComments = props.comments.map((comment, index) => {
+      if (comment.score === 0) {
+        return comment;
+      }
+      if (comment.id === props.comment.id) {
+        comment.score -= 1;
+      }
+      return comment;
+    });
+    props.setComments(newComments);
+  };
+
+  return (
+    <div className="bg-white m-2 flex flex-row p-2" id="comment-container">
+      <div id="score-container" className="bg-gray-100 text-purple-900 m-2 rounded-lg h-[130px] w-[30px] flex flex-col justify-center">
+      
+        <button
+        id={`plus-button-${props.comment.id}`}
+        onClick={() => handlePlus()}
+        className="p-1 m-1"
+        >
+        +
+        </button>
+        <div className="p-1 m-1" id={`score-${props.comment.id}`}>{props.comment.score}</div>
+        <button
+        id={`minus-button-${props.comment.id}`}
+        onClick={() => handleMinus()}
+        className="p-1 m-1"
+        >
+        -
+      </button>
+      </div>   
+
+      <div>
+        <div className="flex p-2 items-center justify-between">
+        <div className="flex p-2 items-center">
+        <img src={props.comment.user.image.png}></img>
+        <h2 className="font-extrabold text-black m-2">{props.comment.user.username}</h2>
+        <h3 className="font-light text-slate-500 text-sm">{props.comment.createdAt}</h3>
+        </div>
+        <div className="flex">
+        <button
+        className="border-2 border-purple-900 text-purple-900 p-1 m-1 flex items-center"
+        id={`reply-button-${props.comment.id}`}
+        onClick={() => {
+          setReplyClicked(true);
+        }}
+        >
+        <p>Reply</p><img src="/icons/icon-reply.svg"/>
+        </button>
+        <Edit id={props.comment.id} content={props.comment.content} commentUser={props.comment.user.username} comments={comments} setComments={props.setComments} currentUser={props.currentUser}/>
+        <Delete />
+        </div>
+      </div>
+      
+      <h1  id={`comment-${props.comment.id}`}>
+        {props.comment.content}
+      </h1>
+
+      <div className="flex p-2">
+        {/* <Edit id={props.comment.id} content={props.comment.content} commentUser={props.comment.user.username} comments={comments} setComments={props.setComments} currentUser={props.currentUser}/>
```

Comment:
reason for this code to be commented out?

Reply:
gone now

Reply:
done